### DISCUSSION TITLE
Is it safe to use the same DiGraph object from multiple threads?

### DISCUSSION THREAD

**Discussion by arvindshmicrosoft** [networkx/networkx#6067] [General]
Not that I have tried this as yet, but I was wondering if there is a global DiGraph object and if multiple threads are calling nx functions / algorithms referencing the same graph (instance), are there any thread safety concerns?

**Comment by MridulS** [comment]
I think it should be fine in majority of cases as majority of algorithms don't update/change the graph object. You can confirm by looking at the functions source code to check if they are manipulating the graph. You could also freeze your graph object with `nx.freeze(G)` to make the graph immutable.

Take this all with a grain of salt, I have never actually tried it 😅
